File: uphsd-cs-molino-jadulos-fritzgerald/MIDTERM2/MachineProblemSet.java
import java.io.*;
import java.util.*;

public class MachineProblemSet {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        // Prompt user for dataset file path
        System.out.print("Enter the dataset file path: ");
        String filePath = sc.nextLine();

        List<String[]> records = new ArrayList<>();

        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line;
            while ((line = br.readLine()) != null) {
                // Parse CSV data by splitting by comma
                records.add(line.split(","));
            }

            if (records.isEmpty()) {
                System.out.println("The file is empty.");
                return;
            }

            // MP02: Display the first 10 rows
            System.out.println("\n--- MP02: First 10 Rows ---");
            for (int i = 0; i < Math.min(11, records.size()); i++) {
                System.out.println(Arrays.toString(records.get(i)));
            }

            // MP10: Detect duplicate records
            System.out.println("\n--- MP10: Duplicate Records ---");
            Set<String> seen = new HashSet<>();
            boolean foundDuplicate = false;
            for (String[] row : records) {
                String rowString = String.join(",", row);
                if (!seen.add(rowString)) {
                    System.out.println("Duplicate found: " + rowString);
                    foundDuplicate = true;
                }
            }
            if (!foundDuplicate) System.out.println("No duplicates detected.");

            // MP20: Convert CSV dataset into JSON format
            System.out.println("\n--- MP20: JSON Format (Preview) ---");
            String[] headers = records.get(0);
            StringBuilder json = new StringBuilder("[\n");
            for (int i = 1; i < records.size(); i++) {
                json.append("  {");
                for (int j = 0; j < headers.length; j++) {
                    String val = j < records.get(i).length ? records.get(i)[j] : "";
                    json.append("\"").append(headers[j]).append("\": \"").append(val).append("\"");
                    if (j < headers.length - 1) json.append(", ");
                }
                json.append("}").append(i == records.size() - 1 ? "" : ",\n");
            }
            json.append("\n]");
            System.out.println(json.toString());

        } catch (IOException e) {
            System.err.println("Error reading file: " + e.getMessage());
        }
    }
}

File: uphsd-cs-molino-jadulos-fritzgerald/MIDTERM2/MachineProblemSet.js
const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

// Prompt user for dataset location
rl.question('Enter the dataset file path: ', (filePath) => {
    try {
        const data = fs.readFileSync(filePath, 'utf8');
        // Parse CSV data into arrays
        const rows = data.split('\n').filter(row => row.trim() !== '').map(row => row.split(','));
        const headers = rows[0];

        // MP02: Display first 10 rows
        console.log('\n--- MP02: First 10 Rows ---');
        console.table(rows.slice(0, 11));

        // MP10: Detect duplicate records
        console.log('\n--- MP10: Duplicate Records ---');
        const seen = new Set();
        let duplicates = [];
        rows.forEach((row, index) => {
            const rowStr = row.join(',');
            if (seen.has(rowStr)) {
                duplicates.push(`Row ${index}: ${rowStr}`);
            }
            seen.add(rowStr);
        });
        console.log(duplicates.length > 0 ? duplicates.join('\n') : 'No duplicates found.');

        // MP20: Convert CSV dataset into JSON format
        console.log('\n--- MP20: JSON Format ---');
        const jsonData = rows.slice(1).map(row => {
            let obj = {};
            headers.forEach((header, i) => {
                obj[header.trim()] = row[i] ? row[i].trim() : "";
            });
            return obj;
        });
        console.log(JSON.stringify(jsonData, null, 2));

    } catch (err) {
        console.error("Error: Could not read file. Check the path.");
    } finally {
        rl.close();
    }
});